lib/config.rb

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'pathname'
+require 'yaml'
+
+require 'symbolize'
+
+# Handles parsing the config directory and the files in it.
+# Config files live by default in the app-level config directory,
+# i.e., rsvp/config.
+# This can be augmented when necessary (for testing, mainly)
+# by setting options[:config_dir] with the -d/--config-dir flag.
+# The -c/--config_profile options are for specialized config files
+# which are merged with the default files.
+# If you pass the option "-c dlxs" then RSVP will try to load the following:
+#   config/config.yml
+#   config/config.local.yml (optional)
+#   config/config.dlxs.yml
+#   config/config.dlxs.local.yml (optional)
+# in order, successively merging them. Meaning that values loaded later
+# override earlier ones.
+# There is no way to remove a config value using this mechanism, so care
+# should be taken to design configurations that do not require the mere
+# presence of a key to function correctly.
+class Config
+  def initialize(options = {})
+    raise 'non-Hash options passed to Config.new' unless options.is_a? Hash
+
+    @options = options
+    config.merge! Symbolize.symbolize(options)
+    @config.freeze
+  end
+
+  def to_s
+    config.to_s
+  end
+
+  def config
+    return @config unless @config.nil?
+
+    @config = Symbolize.symbolize YAML.load_file config_path
+    file = local_config_path
+    if File.exist? file
+      @config.merge! Symbolize.symbolize(YAML.load_file(file) || {})
+    end
+    @config.merge! profile_config if @options.key? :config_profile
+    @config
+  end
+
+  # Delegate Hash methods to @config
+  def respond_to_missing?(method, _include_all)
+    config.respond_to?(method) || super
+  end
+
+  def method_missing(method, *args, &block)
+    config.send(method, *args, &block)
+  end
+
```

Comment:
@moseshll Not exactly understanding why you have respond_to_missing? and method_missing.  Not sure how it is used.  I suspect it's fine but just has to do with something in Ruby I don't know about.

Reply:
`method_missing` is delegating `Hash` methods by forwarding them onto `@config`, so the `Config` object can be used as a Hash from outside. The comment to that effect is hidden be being attached to `respond_to_missing?` -- it's not obvious the two methods belong together, however....

RuboCop made me add `respond_to_missing?`. I thought about doing a `rubocop:disable Style/MissingRespondToMissing` on it, but there are a couple of articles out there about the potential dire consequences of omitting it, for example https://thoughtbot.com/blog/always-define-respond-to-missing-when-overriding

So anyway, that's what's up with that. So you can say `Config.new[:some_key]` or whatever, instead of `Config.new.config[:some_key]`

Reply:
I see...